mllib/src/test/scala/org/apache/spark/mllib/util/MLlibTestSparkContext.scala

```diff
@@ -24,14 +24,18 @@ import org.scalatest.Suite
 
 import org.apache.spark.{SparkConf, SparkContext}
 import org.apache.spark.ml.util.TempDirectory
-import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.{SQLContext, SQLImplicits}
 import org.apache.spark.util.Utils
 
 trait MLlibTestSparkContext extends TempDirectory { self: Suite =>
   @transient var sc: SparkContext = _
   @transient var sqlContext: SQLContext = _
   @transient var checkpointDir: String = _
 
+  protected object testImplicits extends SQLImplicits {
```

Comment:
This is useful to add. But if it is no longer required if we simplify the unit tests above. We can make this change in a follow-up PR.